extensions/amp-font/0.1/amp-font.js

```diff
@@ -0,0 +1,327 @@
+    // Measure until timeout (or font load).
+    let vsyncTask = {
+      measure: () => {
+        if (this.compareMeasurements_()) {
+          this.onFontLoadSuccess_();
+        } else if (this.hasTimedOut_) {
+          this.onFontLoadError_();
+        } else {
+          vsync.run(vsyncTask);
+        }
+      },
+      mutate: () => {}
+    };
+    vsync.run(vsyncTask);
+  }
+
+
+  /**
+   * Step 1 for loading font on browsers that don't support font loading events.
+   * Creates divs hidden from the viewport and measures dimensions for default
+   * fonts.
+   * @private
+   */
+  createElements_() {
+    let containerElement = this.domElements_.CONTAINER =
+        this.document_.createElement('div');
+    style.setStyles(this.domElements_.CONTAINER, {
+      fontSize: '50px',
+      fontVariant: 'normal',
+      left: '-999px',
+      lineHeight: 'normal',
+      margin: 0,
+      padding: 0,
+      position: 'absolute',
+      top: '-999px',
+      visibility: 'hidden'
+    });
+    DEFAULT_FONTS_.forEach(font => {
+      let defaultFontElement = this.document_.createElement('div');
+      this.domElements_.DEFAULT_FONT_ELEMENTS.push(defaultFontElement);
+      defaultFontElement.textContent = TEST_STRING_;
+      style.setStyles(defaultFontElement, {
+        fontFamily: font,
+        margin: 0,
+        padding: 0,
+        whiteSpace: 'nowrap'
+      });
+      containerElement.appendChild(defaultFontElement);
+    });
+    // Adding custom font family to the element to trigger load.
+    // The loading will begin after the container has been appended to the body.
+    let customFontElement = this.domElements_.CUSTOM_FONT_ELEMENT =
+        this.document_.createElement('div');
+    style.setStyles(customFontElement, {
+      fontFamily: this.customFontFamily_,
+      margin: 0,
+      padding: 0,
+      whiteSpace: 'nowrap'
+    });
+    customFontElement.textContent = TEST_STRING_;
+    containerElement.appendChild(customFontElement);
+    this.document_.body.appendChild(containerElement);
+  }
+
+
+  /**
+   * Compare dimensions between elements styled with default fonts and custom
+   * font.
+   * @returns {boolean} Returns true if the dimensions are noticeably different
+   * else returns false.
+   * @private
+   */
+  compareMeasurements_() {
+    let hasWidthChanged = false;
+    let hasHeightChanged = false;
+    this.domElements_.DEFAULT_FONT_ELEMENTS.some(defaultElement => {
+      hasWidthChanged = (
+          Math.abs(
+              defaultElement./*REVIEW*/offsetWidth -
+              this.domElements_.CUSTOM_FONT_ELEMENT./*REVIEW*/offsetWidth) >
+            TOLERANCE_);
+      hasWidthChanged = (
+          Math.abs(
+              defaultElement./*REVIEW*/offsetHeight -
+              this.domElements_.CUSTOM_FONT_ELEMENT./*REVIEW*/offsetHeight) >
+            TOLERANCE_);
+      if (hasWidthChanged || hasHeightChanged) {
+        return true;
+      }
+    });
+    if (hasWidthChanged || hasHeightChanged) {
```

Comment:
you could just return what `some` actually returns which already `true` or `false`